### PR TITLE
feat: add path multiaddr support

### PR DESCRIPTION
**Pull request by jacobheun**

### CODE REVIEW

File: test/listen-dial.spec.js [hunk 1]
```diff
@@ -54,87 +61,84 @@ describe('listen', () => {
     })
   })
 
+  it('listen on path', async () => {
+    // Windows doesn't support unix paths
+    const mh = isWindows
+      ? multiaddr('/ip4/0.0.0.0/tcp/8080')
```
Comment (vasco-santos) [inline test/listen-dial.spec.js]: should we just skip this test in windows instead?

File: test/listen-dial.spec.js [hunk 2]
```diff
@@ -192,6 +196,29 @@ describe('dial', () => {
     await listener.close()
   })
 
+  it('dial on path', async () => {
+    // Windows doesn't support unix paths
+    const ma = isWindows
```
Comment (vasco-santos) [inline test/listen-dial.spec.js]: as before

File: test/listen-dial.spec.js (hunk 1 above)
Comment (jacobheun) [reply test/listen-dial.spec.js]: yeah, that make more sense

File: test/listen-dial.spec.js [hunk 3]
```diff
@@ -192,6 +195,28 @@ describe('dial', () => {
     await listener.close()
   })
 
+  it('dial on path', async () => {
+    // Windows doesn't support unix paths
+    if (isWindows) return this.skip()
```
Comment (vasco-santos) [inline test/listen-dial.spec.js]: you cannot use `this` with an arrow function above

Comment (jacobheun) [reply test/listen-dial.spec.js]: bah! fixed one but not the other. Fixed now.

**Pull request merged**